pkg/sql/show_tenant.go

```diff
@@ -91,12 +94,35 @@ func (n *showTenantNode) startExec(params runParams) error {
 	if err != nil {
 		return err
 	}
+	var job *jobs.Job
 	n.tenantInfo = tenantRecord
 	jobId := n.tenantInfo.TenantReplicationJobID
-	registry := params.p.execCfg.JobRegistry
-	job, err := registry.LoadJobWithTxn(params.ctx, jobId, params.p.Txn())
-	if err != nil {
-		return err
+	if jobId == 0 {
+		n.replicationStatus = n.tenantInfo.State.String()
+	} else {
+		registry := params.p.execCfg.JobRegistry
+		job, err = registry.LoadJobWithTxn(params.ctx, jobId, params.p.Txn())
+		if err != nil {
+			return err
+		}
+		switch n.tenantInfo.State {
+		case descpb.TenantInfo_ADD:
+			switch job.Status() {
+			case jobs.StatusPending:
+				// Replication did not start yet, this is similar to a tenant in the ADD state.
+				n.replicationStatus = n.tenantInfo.State.String()
+			case jobs.StatusRunning, jobs.StatusPauseRequested:
+				n.replicationStatus = "REPLICATING"
```

Comment:
nit: feel free to ignore, but if you want to pull these strings out to the top so it's easier to maintain.